rasa/core/featurizers/single_state_featurizer.py

```diff
@@ -227,26 +217,20 @@ def _get_name_attribute(attributes: Set[Text]) -> Optional[Text]:
     def _extract_state_features(
         self,
         sub_state: SubState,
-        interpreter: NaturalLanguageInterpreter,
         sparse: bool = False,
+        e2e_features: Optional[Dict[Text, Message]] = None,
     ) -> Dict[Text, List["Features"]]:
-        # this method is called during both prediction and training,
-        # `self._use_regex_interpreter == True` means that core was trained
-        # separately, therefore substitute interpreter based on some trained
-        # nlu model with default RegexInterpreter to make sure
-        # that prediction and train time features are the same
-        if self._use_regex_interpreter and not isinstance(
-            interpreter, RegexInterpreter
-        ):
-            interpreter = RegexInterpreter()
+        key = next(
+            k for k in sub_state.keys() if k in {ACTION_NAME, ACTION_TEXT, INTENT, TEXT}
+        )
+        parsed_message = e2e_features[key]
+        assert parsed_message
```

Comment:
Same Problem here:

Was just cherry picking some parts to get the new featurization working without the graph and noticed that this code is wrong. Should probably be something like:
```
            key = next(
                v
                for k, v in message.data.items()
                if k in {ACTION_NAME, ACTION_TEXT, INTENT, TEXT}
            )
            additional_features[key] = message
```
otherwise you are literally saving `additional_features["action_name"] = message` for almost every message

Also I got the case where an substate with `{action_name: action_listen}` would get in there, and that wasn't in `e2e_features` dict at all. For that case I have done this so far: combining the old and the new one. Not sure it's working though. Maybe sth else is wrong

`parsed_message = e2e_features.get(key, Message(data=sub_state))`

Reply:
I think there can be training trackers without `action_listen` as that's added by the `generator`. I'd just add the action_listen part as default then.